src/main/java/duke/database/Storage.java

```diff
@@ -0,0 +1,215 @@
+package duke.database;
+
+import duke.component.Ui;
+import duke.task.Task;
+import duke.exception.DukeException;
+import duke.task.Todo;
+import duke.task.Deadline;
+import duke.task.Event;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * Storage will initialise the save file and obtain the data within
+ * else it will create a new file and allow user to update it.
+ *
+ * @author TeoShyanJie
+ *
+ */
+public class Storage {
+    /** Declaration of file as File type. */
+    private File file;
+
+    /** Ui of the Duke Program. */
+    private Ui ui;
+
+    /**
+     * Storage Constructor.
+     * @param ui Ui of Duke Constructor.
+     */
+    public Storage(Ui ui) {
+        this.ui = ui;
+    }
+
+    /**
+     * To get the project root and path to create the txt file or check if text file is available.
+     */
+    public void initialise() {
```

Comment:
I like the way you designed your Storage class, especially this method to create a new directory if one does not already exist 👍